File: CS50x/Week02/Readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

// Common variables
int i, letterCounter, wordCounter, sentenceCounter;

// Function Prototipes
int get_letters(int lenght, string txt);
int get_words(int lenght, string txt);
int get_sentences(int lenght, string txt);

int main(void)
{
    // Get a text
    string text = get_string("Text: ");
    // Get the text lenght (number of characters) to pass to functions
    int lenght = strlen(text);

    // Count the number of letters, words and sentences
    // Letters
    int nLetters = get_letters(lenght, text);
    // Words
    int nWords = get_words(lenght, text);
    // Sentences
    int nSentences = get_sentences(lenght, text);

    // Coleman-Liau Index
    float L = (nLetters * 100.00) / nWords;
    float S = (nSentences * 100.00) / nWords;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    // Outputs
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", round(index));
    }
}

int get_letters(int lenght, string txt)
{
    for (i = 0; i < lenght; i++)
    {
        if (txt[i] >= 'A' && txt[i] <= 'z')
        {
            letterCounter++;
        }
    }
    return letterCounter;
}

int get_words(int lenght, string txt)
{
    for (i = 0; i < lenght; i++)
    {
        if (txt[i] == ' ')
        {
            wordCounter++;
        }
    }
    return wordCounter + 1;
}

int get_sentences(int lenght, string txt)
{
    for (i = 0; i < lenght; i++)
    {
        if (txt[i] == '.' || txt[i] == '?' || txt[i] == '!')
        {
            sentenceCounter++;
        }
    }
    return sentenceCounter;
}

File: CS50x/Week06/Sentimental-Readability/readability.py
from cs50 import get_string

# Get a text
text = get_string("Text: ")

# Number of words
word = text.strip().split(" ")
n_words = len(word)

# Number of letters
n_letters = 0
for i in range(len(text)):
    if text[i].isalpha():
        n_letters += 1

# Number of sentences
n_sentences = 0
for i in range(len(text)):
    if text[i] == "." or text[1] == "!" or text[i] == "?":
        n_sentences += 1

# Coleman-Liau Index
L = (n_letters * 100.00) / n_words
S = (n_sentences * 100.00) / n_words
index = 0.0588 * L - 0.296 * S - 15.8

# Outputs
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print("Grade ", round(index))
